src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -292,39 +178,31 @@ public override async Task<IEnumerable<string>> EnumerateVersions(PackageURL pur
         {
             try
             {
-                string? packageName = purl?.Name;
+                string? packageName = purl.Name;
+                string? packageVersion = purl.Version;
                 if (packageName == null)
                 {
                     return null;
                 }
-                HttpClient httpClient = CreateHttpClient();
 
-                string? content = await GetHttpStringCache(httpClient, $"{RegistrationEndpoint}{packageName.ToLowerInvariant()}/index.json", useCache);
-                return content;
-            }
-            catch (Exception ex)
-            {
-                Logger.Debug(ex, $"Error fetching NuGet metadata: {ex.Message}");
-                return null;
-            }
-        }
-        
-        public async Task<JsonElement?> GetVersionUriMetadata(string versionUri, bool useCache = true)
-        {
-            try
-            {
-                HttpClient httpClient = CreateHttpClient();
+                // If no package version provided, default to the latest version.
+                if (string.IsNullOrWhiteSpace(packageVersion))
+                {
+                    string latestVersion = await ManagerProvider.GetLatestVersionAsync(purl);
+                    packageVersion = latestVersion;
+                }
 
-                string? content = await GetHttpStringCache(httpClient, versionUri, useCache);
-                if (string.IsNullOrEmpty(content)) { return null; }
+                // Construct a new PackageURL that's guaranteed to have a  version.
+                PackageURL purlWithVersion = new (purl.Type, purl.Namespace, packageName, packageVersion, purl.Qualifiers, purl.Subpath);
+                
+                NuGetMetadata? packageVersionMetadata =
+                    await ManagerProvider.GetMetadataAsync(purlWithVersion, useCache: useCache) as NuGetMetadata;
```

Comment:
Regarding casting the result to `NuGetMetadata` - this goes with my earlier comment about other implementations of `IManagerProvider` not necessarily being suitable for use with this class. If we had a Nuget-specific version of `IManagerProvider`, that would eliminate the need for casting the return value here, since the Nuget-specific `IManagerProvider` would just return `NuGetMetadata` directly.